libgnucash/app-utils/test/gtest-gnc-option.cpp

```diff
@@ -0,0 +1,1350 @@
+{
+    GncOption option{GncOptionValidatedValue<const QofInstance*>{
+        section, name, key, doc_string, (const QofInstance*)value,
+        [](const QofInstance* new_value) -> bool
+            {
+                return GNC_IS_COMMODITY (new_value) &&
+                    gnc_commodity_is_currency(GNC_COMMODITY(new_value));
+            }, is_currency ? GncOptionUIType::CURRENCY : GncOptionUIType::COMMODITY}
+    };
+    return option;
+}
+
+TEST_F(GncOptionCommodityTest, test_currency_ctor)
+{
+    EXPECT_THROW({
+            auto option = make_currency_option("foo", "bar", "baz",
+                                               "Phony Option", m_hpe, false);
+        }, std::invalid_argument);
+    EXPECT_NO_THROW({
+            auto option = make_currency_option("foo", "bar", "baz",
+                                               "Phony Option", m_eur, true);
+        });
+    EXPECT_NO_THROW({
+            auto option = make_currency_option("foo", "bar", "baz",
+                                               "Phony Option", m_usd, true);
+        });
+}
+
+TEST_F(GncOptionCommodityTest, test_currency_setter)
+{
+    auto option = make_currency_option("foo", "bar", "baz", "Phony Option",
+                                      m_eur, true);
+    EXPECT_NO_THROW({
+            option.set_value((const QofInstance*)m_usd);
+        });
+    EXPECT_PRED2(gnc_commodity_equal, m_usd,
+                 GNC_COMMODITY(option.get_value<const QofInstance*>()));
+    EXPECT_THROW({
+            option.set_value((const QofInstance*)m_hpe);
+        }, std::invalid_argument);
+    EXPECT_PRED2(gnc_commodity_equal, m_usd,
+                 GNC_COMMODITY(option.get_value<const QofInstance*>()));
+}
+
+TEST_F(GncOptionCommodityTest, test_currency_validator)
+{
+    auto option = make_currency_option("foo", "bar", "baz", "Phony Option",
+                                      m_eur, true);
+    EXPECT_TRUE(option.validate((const QofInstance*)m_usd));
+    EXPECT_FALSE(option.validate((const QofInstance*)m_aapl));
+}
+
+static inline std::string make_currency_str(gnc_commodity* cur)
+{
+    std::string cur_str{gnc_commodity_get_mnemonic(cur)};
+    return cur_str;
+}
+
+static inline std::string make_commodity_str(gnc_commodity* com)
+{
+    std::string com_str{gnc_commodity_get_namespace(com)};
+    com_str += " ";
+    com_str += gnc_commodity_get_mnemonic(com);
+    return com_str;
+}
+
+static inline std::string make_currency_SCM_str(gnc_commodity* cur)
+{
+    std::string cur_str{gnc_commodity_get_mnemonic(cur)};
+    cur_str.insert(0, "\"");
+    cur_str += "\"";
+    return cur_str;
+}
+
+static inline std::string make_commodity_SCM_str(gnc_commodity* com)
+{
+    std::string com_str{commodity_scm_intro};
+    com_str += "\"";
+    com_str += gnc_commodity_get_namespace(com);
+    com_str += "\" \"";
+    com_str += gnc_commodity_get_mnemonic(com);
+    com_str += "\")";
+    return com_str;
+}
```

Comment:
The above two functions seem (luckily) unused.

Reply:
And other than in the function above, `commodity_scm_intro` isn't used anywhere else either.